File: backend/src/modules/transportation/vehicles/vehicles.controller.ts
import { Body, Controller, Get, Headers, Post, UseGuards } from '@nestjs/common';
import { HeaderAuthGuard, Validation } from '../../../common/auth/header-auth.guard';
import { CurrentUser } from '../../../common/auth/current-user.decorator';
import { VehiclesService } from './vehicles.service';

// Base path /api/Transportation (§6.1 — vehicles & vehicle types). Every action
// requires the CompanyName + UserToken headers via the guard. Filters/pagination
// ride in HTTP headers (PageNo/NoOfItems); deletes/approvals are POSTs with an
// Id header — exactly as documented.
@Controller('api/Transportation')
@UseGuards(HeaderAuthGuard)
export class VehiclesController {
  constructor(private readonly service: VehiclesService) {}

  @Get('getAllVehicleType')
  getAllVehicleType() {
    return this.service.getAllVehicleType();
  }

  @Post('AddVehicleType')
  addVehicleType(@Body() body: { Type: string; Active?: boolean }) {
    return this.service.addVehicleType(body?.Type, body?.Active);
  }

  @Post('UpdateVehicleType')
  updateVehicleType(@Body() body: { Id: number; Type: string; Active?: boolean }) {
    return this.service.updateVehicleType(Number(body?.Id), body?.Type, body?.Active);
  }

  @Get('getAllTransportationVehicle')
  getAllVehicles(@Headers('pageno') pageNo?: string, @Headers('noofitems') noOfItems?: string) {
    return this.service.getAllVehicles(Number(pageNo) || 1, Number(noOfItems) || 20);
  }

  @Post('AddTransportationVehicle')
  addVehicle(
    @Body() body: { VehicleTypeId: number; Capacity: number; Active?: boolean },
    @CurrentUser() v: Validation,
  ) {
    return this.service.addVehicle(Number(body?.VehicleTypeId), Number(body?.Capacity), body?.Active, v.userId);
  }

  @Post('UpdateTransportationVehicle')
  updateVehicle(
    @Body() body: { Id: number; VehicleTypeId: number; Capacity: number; Active?: boolean },
    @CurrentUser() v: Validation,
  ) {
    return this.service.updateVehicle(
      Number(body?.Id),
      Number(body?.VehicleTypeId),
      Number(body?.Capacity),
      body?.Active,
      v.userId,
    );
  }

  @Post('DeleteTransportationVehicle')
  removeVehicle(@Headers('id') id?: string) {
    return this.service.removeVehicle(Number(id));
  }

  @Post('ApproveTransportationVehicle')
  approveVehicle(@Headers('id') id: string, @Headers('approve') approve: string, @CurrentUser() v: Validation) {
    return this.service.approveVehicle(Number(id), String(approve).toLowerCase() === 'true', v.userId);
  }
}

File: backend-dotnet/GarasTransport.Api/Modules/Transportation/Vehicles/VehiclesController.cs
using GarasTransport.Api.Common;
using Microsoft.AspNetCore.Mvc;

namespace GarasTransport.Api.Modules.Transportation.Vehicles;

// Base path /api/Transportation (§6.1 — vehicles & vehicle types).
[Route("api/Transportation")]
[HeaderAuth]
public class VehiclesController : ApiControllerBase
{
    private readonly VehiclesService _service;

    public VehiclesController(VehiclesService service) => _service = service;

    [HttpGet("getAllVehicleType")]
    public Task<object> GetAllVehicleType() => _service.GetAllVehicleType();

    [HttpPost("AddVehicleType")]
    public Task<object> AddVehicleType([FromBody] VehicleTypeBody body) =>
        _service.AddVehicleType(body?.Type, body?.Active);

    [HttpPost("UpdateVehicleType")]
    public Task<object> UpdateVehicleType([FromBody] VehicleTypeBody body) =>
        _service.UpdateVehicleType(body?.Id ?? 0, body?.Type, body?.Active);

    [HttpGet("getAllTransportationVehicle")]
    public Task<object> GetAllVehicles() =>
        _service.GetAllVehicles(HeaderInt("PageNo", 1), HeaderInt("NoOfItems", 20));

    [HttpPost("AddTransportationVehicle")]
    public Task<object> AddVehicle([FromBody] VehicleBody body) =>
        _service.AddVehicle(body?.VehicleTypeId ?? 0, body?.Capacity ?? 0, body?.Active, CurrentUser.UserId);

    [HttpPost("UpdateTransportationVehicle")]
    public Task<object> UpdateVehicle([FromBody] VehicleBody body) =>
        _service.UpdateVehicle(body?.Id ?? 0, body?.VehicleTypeId ?? 0, body?.Capacity ?? 0, body?.Active, CurrentUser.UserId);

    [HttpPost("DeleteTransportationVehicle")]
    public Task<object> RemoveVehicle() => _service.RemoveVehicle(HeaderInt("Id", 0));

    [HttpPost("ApproveTransportationVehicle")]
    public Task<object> ApproveVehicle() =>
        _service.ApproveVehicle(HeaderInt("Id", 0), string.Equals(Header("Approve"), "true", StringComparison.OrdinalIgnoreCase), CurrentUser.UserId);
}

public class VehicleTypeBody
{
    public int? Id { get; set; }
    public string? Type { get; set; }
    public bool? Active { get; set; }
}

public class VehicleBody
{
    public int? Id { get; set; }
    public int? VehicleTypeId { get; set; }
    public int? Capacity { get; set; }
    public bool? Active { get; set; }
}
